lib/rules/array-callback-return.js

```diff
@@ -273,7 +273,8 @@ module.exports = {
                         node,
                         messageId,
                         data: {
-                            name: lodash.upperFirst(astUtils.getFunctionNameWithKind(funcInfo.node))
+                            name: astUtils.getFunctionNameWithKind(funcInfo.node),
+                            arrayMethodName: "Array.prototype.".concat(funcInfo.arrayMethodName)
```

Comment:
This path can also report `Array.from`, e.g.:

```js
/* eslint array-callback-return: error */

// it shouldn't be "Array.prototype.from() expects a return value from arrow function"
Array.from(foo, () => { return; });
```

Maybe we could make a function that gets `funcInfo.arrayMethodName` and returns its "full name", and then use that function here and from the other `context.report`?

Reply:
Thanks, those lines felt fishy and a function is the right answer. I went ahead and included all class methods (of, isArray, from) in case the function was ever lifted somewhere else. Pushed a commit with this change.